site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/CategoryDropdown.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from "react";
+import { PropTypes } from "prop-types";
+import { SelectField } from "react-invenio-forms";
+import _find from 'lodash';
+
+class CategoryDropdown extends Component {
+
+    constructor(props) {
+        super(props);
+        let activeCategory = null;
+        if (Array.isArray(props.categories)) {
+            activeCategory = this.props.categories.find((el) => (el.key === props.defaultCategory))
+        }
+
+        this.state = {
+            activeCategory: activeCategory,
+        }
+    }
+
+    serializeCategory(category) {
+        if (!category) {
+            return null;
+        }
+        return {
+            text: category.title,
+            value: category.key
+        }
+    }
+
+    onChange = ({ data, formikProps }) => {
+        const activeCategory = this.props.categories.find((el) => (el.key === data.value))
+        this.setState({
+            activeCategory: activeCategory
+        });
+        formikProps.form.setFieldValue('category', data.value);
+    }
+
+    render() {
+        const { categories } = this.props;
+
+        const serializedCategories = Array.isArray(categories) ? categories.map((cat) => (this.serializeCategory(cat))) : []
+
+        return (
+            <>
+                <SelectField
+                    options={serializedCategories}
+                    fieldPath='category'
+                    label='Category'
+                    required
+                    width={4}
+                    onChange={this.onChange}
+                />
+                {this.state.activeCategory && <div dangerouslySetInnerHTML={{ __html: this.state.activeCategory.description }}/>}
```

Comment:
`category.description` comes from the backend (`config.py`) and it has some `html` embedded. 

Do we have alternate ways of doing this?

Reply:
this is the only way in react to render HTML...as long as we control this then is fine :)